tests/unit/test_adapter_yaml.py

```diff
@@ -0,0 +1,567 @@
+            """\
+            ---
+            looooooooooooooooooooooooooooooooooooongKey:
+              - item
+            list:
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+            list2: [item, item, item]
+            """
+        )
+        config = YamlfixConfig()
+        config.line_length = 40
+        config.flow_style_sequence = True
+
+        result = fix_code(source, config)
+
+        assert result == fixed_source
+
+    def test_sequence_flow_style_multiline_config(self) -> None:
+        """Make inline list style 'flow-style' configurable even for lists, that would\
+            wrap into the next line."""
+        source = dedent(
+            """\
+            looooooooooooooooooooooooooooooooooooongKey:
+              - item
+            list:
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+            list2:
+              - item
+              - item
+              - item
+            """
+        )
+        fixed_source = dedent(
+            """\
+            ---
+            looooooooooooooooooooooooooooooooooooongKey: [
+              item]
+            list: [loooooooooooooooooooongItem, loooooooooooooooooooongItem,
+              loooooooooooooooooooongItem, loooooooooooooooooooongItem,
+              loooooooooooooooooooongItem, loooooooooooooooooooongItem]
+            list2: [item, item, item]
+            """
+        )
+        config = YamlfixConfig()
+        config.line_length = 40
+        config.flow_style_sequence = True
+        config.flow_style_sequence_multiline = True
+
+        result = fix_code(source, config)
+
+        assert result == fixed_source
+
+    def test_sequence_flow_style_with_trailing_newlines(self) -> None:
+        """Correct flow-style lists that have trailing newlines.
+
+        Without this fix the following block-style list:
+        ```
+        list:
+          - item
+          - item
+
+
+
+        key: value
+        ```
+
+        is converted to this weird-looking flow-style list:
+        ```
+        list: [item, item
+
+
+
+        ]
+        key: value
+        ```
+
+        instead of
+        ```
+        list: [item, item]
+
+
+
+        key: value
+        ```
+        """
+        source = dedent(
+            """\
+            list:
+              - item
+              - item
+
+
+
+            key: value
+            """
+        )
+        fixed_source = dedent(
+            """\
+            ---
+            list: [item, item]
+
+
+
+            key: value
+            """
+        )
```

Comment:
In this case I'd expect the output to be:

```yaml
---
list: [item, item]

key: value
```

Or 

```yaml
---
list: [item, item]
key: value
```

Having many whitelines between elements doesn't look like a good idea. In which cases do you think they're interesting?

Reply:
Recorded in a new issue https://github.com/lyz-code/yamlfix/issues/193